docs/modules/kaolin.ops.spc.rst

```diff
@@ -1,123 +1,134 @@
 .. _kaolin.ops.spc:
 
 kaolin.ops.spc
-==============
+##############
 
 .. _spc:
 
 Structured Point Clouds
------------------------
+***********************
 
-Structured Point Clouds are an octree-base representation of a batch of point cloud hierarchy 
-**points**
+Structured Point Clouds (SPC) is an octree-based representation that is useful to organize and compress geometrically sparse information, like sparse voxelgrids and quantized point clouds. It supports a number of operations, including efficient ray-tracing and convolution.
 
 .. warning::
-  Structured Point Clouds internal layout and structure is still experimental may be modified.
+  Structured Point Clouds internal layout and structure is still experimental and may be modified in the future.
 
-.. _octree-text:
+Octrees
+=======
 
-We represent **octrees** as a :math:`k \times 1` tensor of bytes. 
-Each byte **octrees[i]** represents an octree node corresponding to a 
-:math:`2\times 2\times 2` partition of a cube of space. 
-Each bit indicates the binary occupancy of a child subcube as depicted below. 
+.. _spc_octree:
+
+An `octree <https://en.wikipedia.org/wiki/Octree>`_ is a tree data structure where each node have 8 childrens.
+
+We use this structure to do a recursive three-dimensional space partitioning, 
+i.e: each node represente a partitioning of its 3D space (partition) of :math:`(2, 2, 2)`.
+
+.. figure:: ../img/octree.png
+  :scale: 30 %
+  :alt: An octree 3D partitioning
+
+  Credit: https://en.wikipedia.org/wiki/Octree
+
+Currently Structured Point Clouds are used to represent 3D surfaces, and so all the leaves are at the same ``level`` (depth).
+
+This allow very efficient processing on GPU, with custom CUDA kernels, for ray-tracing and convolution.
+
+The structure is containing more finer details as you go deeper in the tree, below are the Levels 0 through 8 of a structured point cloud teapot model:
+
+.. image:: ../img/spcTeapot.png
+
+We represent ``octrees`` as a tensor of bytes.
```

Comment:
This reads out of order. Here add the high-level part about storing coordinates and then features in the same order. Otherwise, you start with octrees without explaining what an SPC is exactly or what data components it has. 

You can also introduce the container class here.

Reply:
> Here add the high-level part about storing coordinates and then features in the same order.

Technically the SPC can exist without features, like binary voxelgrids or point clouds (although I would agree that in frequent usages that the SPC will be used with features associated to the coordinates, the ray tracing for instance do not necessitate features), which is why:
- I compare the SPC to sparse voxelgrids and quantized point clouds in the first paragraph
- In the `Feature Data` section I wrote "to those coordinates **may** correspond a set of features"
- The high level class do not contain any features.

So I would consider the features to be **associated** to the SPC, but that's just a preference, I can have the features as part of the SPC too. (And so add `features` to `kaolin.rep.Spc` with an associated `level`).

> Otherwise you can start with octrees

The documentation is about the SPC, the octrees being a (the most important) *component* of the SPC, I think it would be a little bit weird to talk about the octree without contextualizing it to the SPC in the documentation for `kaolin.ops.spc.rst`


Following your recommendation, I can only talk about the octree **theoretical structure** + associated features (high level) in the first section (still after the short SPC introduction).

Then in **Related attributes** I can introduce `octrees` and `features` as python attributes.

> You can also introduce the container class here.

I added a short note but I'm considering moving the whole doc to `kaolin.ops.rep`, what do you think?

Reply:
Here's how a doc would read to a new reader:

1. great high-level paragraph, sounds useful
2. "Octrees section" I guess they are providing some background on octrees... Wait, they are also mentioning a variable `octrees`... how does this relate to SPC? Is this SPC itself?

You see what I mean? The reader has to do a lot of unnecessary piecing together, as these components are not introduced. 

**It would be sufficient to add a 2nd paragraph to the very first intro section, something along the lines:**

The SPC data structure is very general. In the SPC data structure, octrees provide a way to store and efficiently retrieve coordinates of points at different levels of the octree hierarchy. It is also possible to associate features to these coordinates using point ordering in memory. Below we detail the low-level representations that comprise SPCs and allow corresponding efficient operations. We also provide a convenience container <LINK> for these low-level attributes.